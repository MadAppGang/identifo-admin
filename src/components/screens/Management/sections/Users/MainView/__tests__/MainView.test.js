import { UsersMainView } from '..';

describe('<UsersMainView />', () => {
  let props;

  beforeEach(() => {
    props = {
      users: [
        {
          id: '1234',
          name: 'John Doe',
          email: 'email@gmail.com',
          latestLogin: 'Never',
          numberOfLogins: 0,
        },
      ],
      fetching: false,
      fetchUsers: jest.fn(),
      history: {
        push: jest.fn(),
      },
    };
  });

  test('renders as expected', () => {
    expect(shallow(<UsersMainView {...props} />)).toMatchSnapshot();
  });

  test('renders as expected with no users', () => {
    props.users = [];
    expect(shallow(<UsersMainView {...props} />)).toMatchSnapshot();
  });

  test('calls fetchUsers prop on mount', () => {
    shallow(<UsersMainView {...props} />);
    expect(props.fetchUsers).toBeCalled();
  });

  test('pushes to history on "create user" button click', () => {
    shallow(<UsersMainView {...props} />).find('Button').simulate('click');
    expect(props.history.push).toBeCalledWith('/management/users/new');
  });

  test('calls fetchUsers prop on search value change', () => {
    shallow(<UsersMainView {...props} />)
      .find('SearchInput')
      .simulate('change', 'value');

    expect(props.fetchUsers).toBeCalledWith({ search: 'value' });
  });

  test('pushes to history on "create user" button (placeholder)', () => {
    props.users = [];

    shallow(<UsersMainView {...props} />)
      .find('UsersPlaceholder')
      .simulate('createNewUserClick');

    expect(props.history.push).toBeCalledWith('/management/users/new');
  });
});
