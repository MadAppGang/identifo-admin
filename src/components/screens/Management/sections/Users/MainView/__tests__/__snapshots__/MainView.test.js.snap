// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersMainView /> renders as expected 1`] = `
<section
  className="iap-management-section"
>
  <p
    className="iap-management-section__title"
  >
    Users
    <Button
      Icon={[Function]}
      disabled={false}
      error={false}
      iconClassName=""
      onClick={[Function]}
      stretch={false}
      transparent={false}
      type="button"
    >
      Create User
    </Button>
  </p>
  <p
    className="iap-management-section__description"
  >
    Look for users, edit, delete them and add new ones.
  </p>
  <SearchInput
    onChange={[Function]}
    placeholder="Search for users"
    timeout={400}
  />
  <UserList
    loading={false}
    users={
      Array [
        Object {
          "email": "email@gmail.com",
          "id": "1234",
          "latestLogin": "Never",
          "name": "John Doe",
          "numberOfLogins": 0,
        },
      ]
    }
  />
</section>
`;

exports[`<UsersMainView /> renders as expected with no users 1`] = `
<section
  className="iap-management-section"
>
  <UsersPlaceholder
    onCreateNewUserClick={[Function]}
  />
</section>
`;
